reactDashboard/src/components/DataCharts/DataCharts.jsx

```diff
@@ -137,95 +203,176 @@ const DataCharts = ({ healthBoardDataset = null }) => {
+        (existingMap) => new Map([...existingMap, ...regionDailyCasesMap])
+      );
+      setDailyDeathsSeriesData(
+        (existingMap) => new Map([...existingMap, ...regionDailyDeathsMap])
+      );
+      setTotalCasesSeriesData(
+        (existingMap) => new Map([...existingMap, ...regionTotalCasesMap])
+      );
+      setTotalDeathsSeriesData(
+        (existingMap) => new Map([...existingMap, ...regionTotalDeathsMap])
+      );
+      setPopulationMap(
+        (existingMap) => new Map([...existingMap, ...populationMap])
+      );
+    }
+  }, [councilAreaDataset]);
+
+  useEffect(() => {
+    setPopulationProportionMap(calculatePopulationProportionMap(populationMap));
+  }, [populationMap]);
+
   useEffect(() => {
     if (healthBoardDataset != null) {
-      const parseDateRange = getNhsCsvDataDateRange(healthBoardDataset);
+      const parseDateRange = getNhsCsvDataDateRange(
+        healthBoardDataset,
+        councilAreaDataset
+      );
       setMaxDateRange(parseDateRange);
       setDateRange(parseDateRange);
     }
-  }, [healthBoardDataset]);
+  }, [healthBoardDataset, councilAreaDataset]);
 
-  // Stop audio on chart or dateRange change
   useEffect(() => {
-    stopAudio();
-  }, [chartType, dateRange]);
+    function getAverageSeriesData(seriesData, regionCode) {
+      if (regionCode == null) {
+        return seriesData.get(FEATURE_CODE_SCOTLAND);
+      }
+      const populationProportion = populationProportionMap.get(regionCode);
+      const scotlandData = seriesData.get(FEATURE_CODE_SCOTLAND);
+      if (scotlandData == null) {
+        return null;
+      }
+      const scaledSeries = scotlandData.map(({ t, y }) => {
+        return {
+          t: t,
+          y: y * populationProportion,
+        };
+      });
+      return scaledSeries;
+    }
 
-  useEffect(() => {
-    const DATASET_COLOUR = "#ec6730";
-
-    function percentageCasesChartConfiguration() {
-      const datasets = [
-        datasetConfiguration(
-          percentageCasesDatasetLabel,
-          percentageCasesSeriesData,
-          DATASET_COLOUR
-        ),
-      ];
-      const configuration = commonChartConfiguration(datasets, dateRange);
+    const REGION_DATASET_COLOUR = "#ec6730";
+    const AVERAGE_DATASET_COLOUR = "blue";
 
-      configuration.options.scales.yAxes[0].ticks.callback = (
-        value,
-        index,
-        values
-      ) => {
-        return Math.round(value) + "%";
-      };
-      configuration.options.tooltips = {
-        backgroundColor: "White",
-        titleFontColor: "#000000",
-        bodyFontColor: "#000000",
-        borderColor: "DarkGray;",
-        borderWidth: 1,
-        position: "nearest",
+    function setChart(
+      datasetLabel,
+      seriesData,
+      regionCode,
+      additionalConfiguration,
+      sonificationLabel
+    ) {
+      const datasets = [];
+      const currentSeriesData = seriesData.get(regionCode);
+      if (currentSeriesData != undefined) {
```

Comment:
Line 321:29:  Expected '!==' and instead saw '!='

Reply:
Fixed, ta